src/track/track.h

```diff
@@ -78,14 +78,16 @@ class Track : public QObject {
     Q_PROPERTY(QString titleInfo READ getTitleInfo STORED false NOTIFY infoChanged)
     Q_PROPERTY(QDateTime sourceSynchronizedAt READ getSourceSynchronizedAt STORED false)
 
-    mixxx::FileAccess getFileAccess() const {
-        // Copying QFileInfo is thread-safe due to implicit sharing,
+    SecurityTokenPointer getFileAccessToken() const {
+        // Copying a QSharedPointer is thread-safe
         // i.e. no locking needed.
-        return m_fileAccess;
+        return m_fileAccess.token();
     }
+
     mixxx::FileInfo getFileInfo() const {
-        // Copying QFileInfo is thread-safe due to implicit sharing,
+        // Copying mixxx::FileInfo based on QFileInfo is thread-safe due to implicit sharing,
         // i.e. no locking needed.
+        static_assert(sizeof(QSharedDataPointer<int>) == sizeof(mixxx::FileInfo));
```

Comment:
while I appreciate the usage of the `static_assert`, I'm not sure the way its implemented is a good idea. It's very leaky. 
I think it would be better if there was a `constexpr static` member on `mixxx::FileInfo` that indicated whether it only consisted of implicitly shared types. Ideally Qt had some machinery for this, but I don't think it does.

Reply:
I have put it here because it proves the comment just in the next line.  If one will add members to FileInfo it will fail here where the mutex lock is required.

Reply:
Right, but it will not fail if the implementation changes but it happens to have the same size. That's why I proposed to have the evaluation of the property in the fileaccess class, but leave the static_assert here. It's just shifting code around a bit in order to not break encapsulation.

Reply:
Now we have a static isQFileInfo(). I was attempted to call it isThreadSafeCopyable() but that would be very confusing in a Info class describing file actions.

Reply:
>  I was attempted to call it isThreadSafeCopyable() but that would be very confusing in a Info class describing file actions.

Yeah, but its a naming issue. The entire point of the static is not to expose the internal QFileInfo imo. I think it sufficies to use a better name or doc comment.